src/language-js/print/assignment.js

```diff
@@ -224,6 +232,16 @@ function isSimpleCall(node) {
   );
 }
 
+function isObjectPropertyWithShortKey(node, options) {
+  return (
+    (node.type === "ObjectProperty" || node.type === "Property") &&
+    node.key &&
+    !node.computed &&
+    node.key.type === "Identifier" &&
+    node.key.name.length <= options.tabWidth + 2
```

Comment:
Why `2`? :smile:

Reply:
It was 3 initially. :D Suggestions?

Reply:
There has to be a sufficient visual overlap for the line break to look natural. I'm having trouble with determining how long exactly it needs to be.

Reply:
Honestly, I also have no ideas, magic numbers scare me :smile: , maybe we just need to make cases with `2`/`3`/`4` values and look where it will be better

Reply:
Added that to snapshots. Have a look.

Reply:
I guess the value 2 was based on this comment https://github.com/prettier/prettier/issues/3711#issuecomment-357245293

Reply:
A sample diff: https://github.com/prettier/prettier-regression-testing/issues/32#issuecomment-780563517